diff --git a/dist/index.mjs b/dist/index.mjs
index c18069c87a507d2fc323d4b633f3941594c9cd28..2171924b870898555072c06be76c0711dc7e5fb1 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -4,13 +4,17 @@ export { ShikiTwoslashError, defaultCompletionIcons, defaultCustomTagIcons, defa
 import '@shikijs/core';
 
 function transformerTwoslash(options = {}) {
+  let twoslasher;
   return createTransformerFactory(
-    createTwoslasher({
-      cache: options?.cache,
-      compilerOptions: {
-        moduleResolution: 100
-      }
-    }),
+    (...args) => {
+      twoslasher ??= createTwoslasher({
+        cache: options?.cache,
+        compilerOptions: {
+          moduleResolution: 100
+        }
+      });
+      return twoslasher(...args);
+    },
     rendererRich(options.rendererRich)
   )(options);
 }
